subprojects/platform-base/src/main/java/org/gradle/language/base/plugins/LifecycleBasePlugin.java

```diff
@@ -55,12 +55,12 @@ private void addClean(final ProjectInternal project) {
         final BuildOutputCleanupRegistry buildOutputCleanupRegistry = project.getServices().get(BuildOutputCleanupRegistry.class);
         buildOutputCleanupRegistry.registerOutputs(buildDir);
 
-        final Provider<Delete> clean = project.getTasks().register(CLEAN_TASK_NAME, Delete.class, cleanTask -> {
+        project.getTasks().register(CLEAN_TASK_NAME, Delete.class, cleanTask -> {
             cleanTask.setDescription("Deletes the build directory.");
             cleanTask.setGroup(BUILD_GROUP);
             cleanTask.delete(buildDir);
+            buildOutputCleanupRegistry.registerOutputs(cleanTask.getTargetFiles());
         });
-        buildOutputCleanupRegistry.registerOutputs(clean.map(cl -> cl.getTargetFiles()));
```

Comment:
The registration should happen even if the clean task is not realized.

Reply:
Right, because stale output cleanup happens on all tasks, not only if/when clean tasks are present in the task graph.

Reply:
CI was happy with the change though, meaning we probably miss coverage in this area. I'll leave this in your capable hands and revert the change in this PR.

Reply:
FTR the motivation for this change was to prevent force-creating clean tasks when instant execution captures registered task outputs. See failures in https://builds.gradle.org/viewLog.html?buildId=33734270&tab=queuedBuildOverviewTab

Reply:
I ended up reworking the failing test to accept the fact that instant execution forces the creation of the clean task if registered.

Reply:
I don't see a way around realizing the `clean` task to obtain the necessary information.